packages/math-extras/src/pointExtras.ts

```diff
@@ -0,0 +1,419 @@
+     */
+    magnitude(): number
+    {
+        return Math.sqrt((this.x * this.x) + (this.y * this.y));
+    },
+
+    /**
+     * Computes the square magnitude of this point.
+     * If you are comparing the lengths of vectors, you should compare the length squared instead
+     * as it is slightly more efficient to calculate.
+     *
+     * Defined as the sum of the squares of each component.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method magnitudeSquared
+     * @memberof PIXI.Point#
+     * @returns {number} The magnitude squared (length squared) of the vector.
+     */
+    /**
+     * Computes the square magnitude of this point.
+     * If you are comparing the lengths of vectors, you should compare the length squared instead
+     * as it is slightly more efficient to calculate.
+     *
+     * Defined as the sum of the squares of each component.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method magnitudeSquared
+     * @memberof PIXI.ObservablePoint#
+     * @returns {number} The magnitude squared (length squared) of the vector.
+     */
+    magnitudeSquared(): number
+    {
+        return (this.x * this.x) + (this.y * this.y);
+    },
+
+    /**
+     * Computes vector projection of `this` on `onto`.
+     *
+     * Imagine a light source, parallel to `onto`, above `this`.
+     * The light would cast rays perpendicular to `onto`.
+     * `this.project(onto)` is the shadow cast by `this` on the line defined by `onto` .
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method project
+     * @memberof PIXI.Point#
+     * @param {IPointData} onto - A non zero vector describing a line on which to project `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `this` on `onto` projection.
+     */
+    /**
+     * Computes vector projection of `this` on `onto`.
+     *
+     * Imagine a light source, parallel to `onto`, above `this`.
+     * The light would cast rays perpendicular to `onto`.
+     * `this.project(onto)` is the shadow cast by `this` on the line defined by `onto` .
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method project
+     * @memberof PIXI.ObservablePoint#
+     * @param {IPointData} onto - A non zero vector describing a line on which to project `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `this` on `onto` projection.
+     */
+    project<T extends IPointData>(onto: IPointData, outPoint?: T): T
+    {
+        if (!outPoint)
+        {
+            (outPoint as any) = new Point();
+        }
+        // Math says: a Projected over b = [(a·b) / (b·b)] * b;
```

Comment:
```suggestion
        // Math says: a Projected over b = [(a·b) / (b·b)] * b; We normalize that to the length of onto.
```

Reply:
The end result is not normalized. We just moved around some parameters to avoid 2 `sqrt` calculations that would cancel each other.

![image](https://user-images.githubusercontent.com/15253010/125388467-e81fa980-e375-11eb-95e3-c301b6165b4e.png)

(see your other conversation for the long explanation, I  don't know how to tag it, sorry 😅 )